runner/llamarunner/runner.go

```diff
@@ -702,15 +647,19 @@ func (s *Server) completion(w http.ResponseWriter, r *http.Request) {
 				flusher.Flush()
 			} else {
 				// Send the final response
-				if err := json.NewEncoder(w).Encode(&CompletionResponse{
-					Stop:         true,
-					StoppedLimit: seq.doneReason == "limit",
-					Timings: Timings{
-						PromptN:     seq.numPromptInputs,
-						PromptMS:    float64(seq.startGenerationTime.Sub(seq.startProcessingTime).Milliseconds()),
-						PredictedN:  seq.numDecoded,
-						PredictedMS: float64(time.Since(seq.startGenerationTime).Milliseconds()),
-					},
+				doneReason := "stop"
+				if seq.doneReason == "limit" {
+					doneReason = "length"
+				}
+				promptDur := common.ParseDurationMs(float64(seq.startGenerationTime.Sub(seq.startProcessingTime).Milliseconds()))
+				evalDur := common.ParseDurationMs(float64(time.Since(seq.startGenerationTime).Milliseconds()))
```

Comment:
Am I reading this right? This first formats a `time.Duration` into milliseconds then parses it back to a `time.Duration`